src/EFCore.PG/ValueGeneration/UUid7ValueGenerator.cs

```diff
@@ -0,0 +1,59 @@
+using System.Runtime.InteropServices;
+
+namespace Npgsql.EntityFrameworkCore.PostgreSQL.ValueGeneration;
+
+/// <summary>
+///     Generates sequential <see cref="Guid" /> values according to the UUID version 7 specification.
+///     Will be updated to use Guid.CreateVersion7 when available.
+/// </summary>
+public class UUid7ValueGenerator : ValueGenerator<Guid>
+{
+    private const byte Variant10xxValue = 0x80;
+    private const ushort Version7Value = 0x7000;
+    private const ushort VersionMask = 0xF000;
+    private const byte Variant10xxMask = 0xC0;
+
+    /// <summary>
+    ///     Gets a value to be assigned to a property.
+    /// </summary>
+    /// <param name="entry">The change tracking entry of the entity for which the value is being generated.</param>
+    /// <returns>The value to be assigned to a property.</returns>
+    public override Guid Next(EntityEntry entry)
+    {
+        Span<byte> guidBytes = stackalloc byte[16];
+        var succeeded = Guid.NewGuid().TryWriteBytes(guidBytes);
+        var unixms = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();
+        Span<byte> counterBytes = stackalloc byte[sizeof(long)];
```

Comment:
**Issue:** The name `counterBytes` seems a bit misleading for the timestamp component, esp. now that we have no counter at all.

Reply:
Can easily change that. Besides if we use your suggestion it automatically falls away

Reply:
Different implementation used which doesn't have the above code, so resolved